test/powershell/engine/Module/ModuleCmdlet.Tests.ps1

```diff
@@ -0,0 +1,506 @@
+            $module.ExportedFunctions.ContainsKey("Test-SecondModuleFunction") | Should -BeExactly $true
+        }
+
+        It "Re-imports modules from file with new members when -Force is used with -AsCustomObject" {
+            $modItem = New-SimpleModule
+            $module = Import-Module $modItem.PSPath -PassThru
+
+            $module.ExportedFunctions.Count | Should -BeExactly 1
+            $module.ExportedFunctions.ContainsKey("Test-FirstModuleFunction") | Should -BeExactly $true
+            $module.ExportedFunctions.ContainsKey("Test-SecondModuleFunction") | Should -BeExactly $false
+
+            New-Item -Force -Path $modItem.PSPath -Value (((Get-Content $modItem) | Out-String) + "`nfunction Test-SecondModuleFunction { Write-Output 'SECONDSTRING' }")
+
+            $module = Import-Module $modItem.PSPath -AsCustomObject -Force
+            $module.'Test-SecondModuleFunction'() | Should -BeExactly "SECONDSTRING"
+        }
+    }
+
+    Context "Classes in modules" {
+        It "Uses updated class definitions when the module is reloaded" {
+            $content = @'
+$passedArgs = $Args
+class Root { $passedArgs = $passedArgs }
+function Get-PassedArgsRoot { [Root]::new().passedArgs }
+function Get-PassedArgsNoRoot { $passedArgs }
+'@
+
+            $rootModName = "rootMod"
+            $rootModDir = "$TestDrive\$rootModName"
+            New-Item -Path $rootModDir -ItemType Directory
+            New-Item -Path $rootModDir -Name "$rootModName.psm1" -Value $content
+
+            Import-Module $rootModDir -ArgumentList 'value1'
+            $rootVal = Get-PassedArgsRoot
+            $noRootVal = Get-PassedArgsNoRoot
+            $rootVal | Should -BeExactly $noRootVal
+
+            Import-Module $rootModDir -ArgumentList 'value2'
+            $rootVal = Get-PassedArgsRoot
+            $rootNoVal = Get-PassedArgsNoRoot
+            $rootVal | Should -BeExactly $noRootVal
+        }
+
+    }
+
+    Context "Parameter handling" {
+        BeforeAll {
+            $validationTests = @(
+                @{ paramName="Variable" },
+                @{ paramName="Function" },
+                @{ paramName="Cmdlet" },
+                @{ paramName="Variable" }
+            )
+
+            $ps = [powershell]::Create()
+        }
+
+        AfterAll {
+            $ps.Dispose()
+        }
+
+        AfterEach {
+            $ps.Streams.ClearStreams()
+        }
+
+        It "Validates a null -<paramName> parameter" -TestCases $validationTests {
+            param($paramName)
+
+            $modItem = New-SimpleModule
+            try
+            {
+                $ps.AddScript("Import-Module $($modItem.PSPath) -$paramName `$null")
+                $ps.Invoke()
+            }
+            catch
+            {
+                $_.FullyQualifiedErrorId | Should -BeExactly "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.ImportModuleCommand"
```

Comment:
What do the test check? Exception? If so we should use  `{ ... } | Should -Throw -ErrorId " ... "`